torch/_dynamo/optimizations/distributed.py

```diff
@@ -236,42 +240,61 @@ def forward(self, *args):
                         return x
 
                 unwrap_singleton_tuple = False
-                for sn in submod.graph.nodes:
+                for sn in input_mod.graph.nodes:
                     if sn.op == "output":
                         if not isinstance(sn.args[0], tuple):
                             unwrap_singleton_tuple = True
                             sn.args = (sn.args,)
-                submod.recompile()
 
+                input_mod.recompile()
                 wrapper = WrapperModule(
-                    self.compiler(submod, args),
+                    self.compiler(input_mod, args),
                     unwrap_singleton_tuple,
                 )
                 return wrapper
 
             def run_node(self, n: Node) -> Any:
                 with fx_traceback.append_stack_trace(n.stack_trace):
                     args, kwargs = self.fetch_args_kwargs_from_env(n)
+                    new_args = []
+                    assert fake_mode
+                    for arg in args:
+                        if isinstance(arg, torch.Tensor) and not isinstance(
+                            arg, torch._subclasses.FakeTensor
+                        ):
+                            new_args.append(fake_mode.from_tensor(arg))
+                        else:
+                            new_args.append(arg)
```

Comment:
I kept meaning to rewrite this as
```suggestion
                    def convert(arg):
                        if isinstance(arg, torch.Tensor) and not isinstance(
                            arg, torch._subclasses.FakeTensor
                        ):
                            return arg
                        return arg
                     
                    new_args = [convert(arg) for arg in args]
```

We also dot this in enough places that a util my be warranted...